power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -665,8 +673,14 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
                 if (tap_changed) {
                     break;
                 }
+                iterations_per_rank[++rank_index] = 0;
             }
             if (tap_changed) {
+                if (static_cast<unsigned long>(++iterations_per_rank[rank_index]) >
+                    2 * max_tap_ranges_per_rank[rank_index]) {
+                    throw MaxIterationReached{
+                        "TapPositionOptimizer::iterate, maximum iterations reached, no solution."};
```

Comment:
it may be useful to add the `maximum iterations reached` to the constructor of the exception. that will also help when making the python exception parser